src/sentry/api/authentication.py

```diff
@@ -550,3 +551,120 @@ def authenticate_token(self, request: Request, token: str) -> tuple[Any, Any]:
         sentry_sdk.get_isolation_scope().set_tag("rpc_auth", True)
 
         return (AnonymousUser(), token)
+
+
+def compare_service_signature(
```

Comment:
pretty much 1:1 rip of https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/seer_rpc.py#L60

Reply:
I'm pretty sure the seer implementation was derived from the cross-region rpc auth. We should consolidate these implementations now that we have 3 of them, but that can happen after this merges.